src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,140 +1,381 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
-import java.util.ListIterator;
+import java.util.*;
 
 public class StudentList implements List<Student> {
+    private int capacity;
+    private int size = 0;
+    private Student[] array;
 
+    public StudentList() {
+        this.capacity = 10;
+        this.array = new Student[this.capacity];
+    }
+
+    public StudentList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            this.capacity = initialCapacity;
+            this.array = new Student[this.capacity];
+        } else {
+            if (initialCapacity != 0) {
+                throw new IllegalArgumentException();
+            }
+
+            this.capacity = 10;
+            this.array = new Student[this.capacity];
+        }
+
+    }
+
+    public StudentList(Collection<? extends Student> collection) {
+        if (collection == null) {
+            throw new NullPointerException();
+        } else {
+            this.size = collection.size();
+            this.capacity = this.size * 2;
+            this.array = new Student[this.capacity];
+            Iterator var2 = collection.iterator();
```

Comment:
Can this be final ?

Reply:
Use more understandable name than `var2`

Reply:
Fixed!